src/services/services.ts

```diff
@@ -316,6 +320,14 @@ namespace ts {
 
             return this.documentationComment;
         }
+
+        getJsDocTags(): JSDocTagInfo[] {
+            if (this.tags === undefined) {
+                this.tags = JsDoc.getJsDocTagsFromDeclarations(this.declarations);
+            }

```

Comment:
Here too the check would need to change to avoid empty array allocation for the default no-JSDoc case.
(to mark the difference between un-computed value and empty value in `this.tags` -- `undefined` versus `null` can be used)